lib/modules/@apostrophecms/login/src/apos/components/TheApostropheLogin.vue

```diff
@@ -1,50 +1,117 @@
-        <button type="submit">Log In</button>
-      </fieldset>
-    </form>
+  <div class="apos-login apos-theme-dark">
+    <div class="apos-login__overlay"></div>
+    <div class="apos-login__menu-overlay"></div>
+    <AposLoginBackground />
+
+    <div class="apos-login__header">
+      <label
+        class="apos-login__project apos-login__project-env"
+        :class="[`apos-login__project-env--${env}`]"
+      >
+        {{ env }}
+      </label>
+      <label class="apos-login__project apos-login__project-name">{{ projectName }}</label>
+      <label class="apos-login--error">{{ error }}</label>
+    </div>
+
+    <div class="apos-login__body">
+      <form>
+        <AposInputString
+          @input="fill($event, 'usernameField')"
+          :field="usernameField.field"
+          :status="usernameField.status"
+          :value="usernameField.value"
+          :modifiers="['dark']"
+        />
+        <AposInputString
+          @input="fill($event, 'passwordField')"
+          :field="passwordField.field"
+          :type="passwordField.type"
+          :status="passwordField.status"
+          :value="passwordField.value"
+          :modifiers="['dark']"
+        />
+        <a href="#" class="apos-login__link">Forgot Password</a>
+        <AposButton
+          class="apos-field--short"
+          :busy="busy"
+          :disabled="busy"
+          type="primary"
+          label="Login"
+          :modifiers="['gradient-on-hover']"
+          @click="submit"
+        />
+      </form>
+    </div>
+
+    <div class="apos-login__footer">
+      <AposLogo class="apos-login__logo"/>
+      <label class="apos-login__logo-name">ApostropheCMS</label>
+      <label class="apos-login__project-version">Version {{ version }}</label>
+    </div>
   </div>
 </template>
 
 <script>
-
 export default {
   name: 'TheApostropheLogin',
   data() {
     return {
-      username: '',
-      password: '',
-      error: false
+      error: '',
+      busy: false,
+      usernameField: {
+        field: {
+          name: 'username',
+          placeholder: 'Enter username',
+          label: 'Username',
+          required: true
+        },
+        status: {},
+        value: { data: '' }
+      },
+      passwordField: {
+        field: {
+          name: 'password',
+          type: 'password',
+          placeholder: 'Enter password',
+          label: 'Password',
+          required: true
+        },
+        status: {},
+        value: { data: '' }
+      },
+      env: apos.context.env,
+      version: apos.context.version,
+      projectName: apos.context.name.replace(/-/g, ' ')
     };
   },
   methods: {
+    fill(value, field) {
+      this[field].value = value
+    },
     async submit() {
       const self = this;
+      self.busy = true;
       self.error = false;
       try {
+        if (!self.usernameField.value.data || !self.passwordField.value.data) {
+          throw new Error('Username and password required.');
+        }
```

Comment:
You could use an AposSchema component to avoid reimplementing validation etc. I think that is probably a good idea.